core/src/main/java/org/apache/calcite/rel/rules/materialize/MaterializedViewAggregateRule.java

```diff
@@ -910,6 +942,53 @@ protected SqlFunction getFloorSqlFunction(TimeUnitRange flag) {
     return Pair.of(resultTopViewProject, requireNonNull(resultViewNode, "resultViewNode"));
   }
 
+  /**
+   * If the view contains a FLOOR(col) and the query contains a range predicate on the col then
+   * rewrite the view to include a predicate based on the query predicate so that the view
+   * can be used.
+   * @return pair of view and added view predicate, or null if the rewrite can't be done
+   */
+  @Override public @Nullable  Pair<RelNode, RexNode> rewriteInputView(RelOptRuleCall call,
```

Comment:
Remove extra whitespace
```suggestion
  @Override public @Nullable Pair<RelNode, RexNode> rewriteInputView(RelOptRuleCall call,
```

Reply:
done